cmake/XLink.cmake

```diff
@@ -14,6 +14,14 @@ if(WIN32)
     set(XLINK_PLATFORM_INCLUDE ${XLINK_ROOT_DIR}/src/pc/Win/include)
 
     file(GLOB XLINK_PLATFORM_SRC "${XLINK_ROOT_DIR}/src/pc/Win/src/*.c")
+
+    if(MINGW)
+        # Remove win_pthread.c, win_semaphore.c, and win_synchapi.c (use MinGW impl.)
```

Comment:
Just wondering, MinGW can be used with Windows threads as well right (different compiler configuration)? In that case should similar check as in source code happen here as well?

Reply:
It is my understanding that MinGW is using Windows threads underneath. Its objective is to provide a compatibility layer to GCC/Linux projects, but platform specific functionality ends up in native Windows DLLs. One could use Windows threads directly, but that would defeat the purpose... I'd considered experimenting with using MSYS2/mingw's [libusb](https://packages.msys2.org/package/mingw-w64-x86_64-libusb) implementation and omitting win_usb.c/h, but that was not breaking.

Reply:
I meant the API (eg. not providing the pthread one), altough I could be wrong. Just saw the option when installing MinGW a while ago.

Regarding getting rid of win_usb.c/h - its in the works. I've got a port of libusb with CMake support which XLink will integrate directly. Supposedly works with all OSes, so no need for distinctions anymore